### PR TITLE
Add Microsoft.AspNetCore.Server.IntegrationTesting.IIS package

### PR DESCRIPTION
**Pull request by pakrym**
Copy IISExpressDeployer into it.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        /// <summary>
+        /// Creates a deployer instance based on settings in <see cref="DeploymentParameters"/>.
+        /// </summary>
+        /// <param name="deploymentParameters"></param>
+        /// <param name="loggerFactory"></param>
+        /// <returns></returns>
+        public static ApplicationDeployer Create(DeploymentParameters deploymentParameters, ILoggerFactory loggerFactory)
+        {
+            if (deploymentParameters == null)
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs]:
```c#
deploymentParameters ?? throw new ArgumentNullException(nameof(deploymentParameters));
```

File: test/Common.FunctionalTests/Utilities/FunctionalTestsBase.cs [hunk 2]
```diff
@@ -25,17 +25,13 @@ public FunctionalTestsBase(ITestOutputHelper output = null) : base(output)
...
-                _deployer = new IISDeployer(parameters, LoggerFactory);
-            }
-            else if (parameters.ServerType == ServerType.IISExpress)
+            if (parameters.ServerType == ServerType.IISExpress)
             {
                 parameters.ServerConfigTemplateContent = parameters.ServerConfigTemplateContent ?? File.ReadAllText("IISExpress.config");
-                _deployer = new IISExpressDeployer(parameters, LoggerFactory);
             }
 
+            _deployer = IISApplicationDeployerFactory.Create(parameters, LoggerFactory);
```
Comment (jkotalik) [inline test/Common.FunctionalTests/Utilities/FunctionalTestsBase.cs]: Add this to the fixture?

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public static ApplicationDeployer Create(DeploymentParameters deploymentParamete
             switch (deploymentParameters.ServerType)
             {
                 case ServerType.IISExpress:
-                    return new IISExpressDeployer(deploymentParameters, loggerFactory);
+                    return new IIS.IISExpressDeployer(deploymentParameters, loggerFactory);
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs]: 👍

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs (hunk 1 above)
Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs]: It's against my beliefs.

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Microsoft.AspNetCore.Server.IntegrationTesting.IIS.csproj [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <PackageId>Microsoft.AspNetCore.Server.IIS</PackageId>
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Microsoft.AspNetCore.Server.IntegrationTesting.IIS.csproj]: Wrong package id and description

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Microsoft.AspNetCore.Server.IntegrationTesting.IIS.csproj [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    <Description>Provides support for hosting ASP.NET Core in IIS using the AspNetCoreModule.</Description>
+    <NoWarn>$(NoWarn);CS1591</NoWarn>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PackageTags>aspnetcore;iis</PackageTags>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Server.IntegrationTesting" PrivateAssets="None" Version="$(MicrosoftAspNetCoreServerIntegrationTestingPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" PrivateAssets="All" Version="$(MicrosoftExtensionsCommandLineUtilsSourcesPackageVersion)" />
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Microsoft.AspNetCore.Server.IntegrationTesting.IIS.csproj]: CommandLine?

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs (hunk 1 above)
Comment (jkotalik) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs]: I'll just fix it in a follow up PR 😄

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/DotNetCommands.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    // Copied from hosting
+    // TODO: make public while removing IISExpressDeployer
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/DotNetCommands.cs]: Follow up issue in hosting?

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    /// </summary>
+    public class IISApplicationDeployerFactory
+    {
+        /// <summary>
+        /// Creates a deployer instance based on settings in <see cref="DeploymentParameters"/>.
+        /// </summary>
+        /// <param name="deploymentParameters"></param>
+        /// <param name="loggerFactory"></param>
+        /// <returns></returns>
+        public static ApplicationDeployer Create(DeploymentParameters deploymentParameters, ILoggerFactory loggerFactory)
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs]: The name IISApplicationDeployerFactory bugs me but I don't have a better name. What if you made it an extension method on DeploymentParameters instead? Then you wouldn't see the name, and it'd be shorter.
`deploymentParameters.CreateServer(loggerFactory)`

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Http.config [hunk 8]
```diff
@@ -0,0 +1,1034 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Http.config]: Why is this here? Use IIS.config or IISExpress.config.

Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Http.config]: I don't want to break any downlevel dependencies by having a different config file.

Comment (jkotalik) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/Http.config]: ic
